root/layout/components/Footer.js

```diff
@@ -9,87 +9,568 @@
 
 import * as React from 'react';
 
-import {CatalystContext} from '../../context';
-import DBDefs from '../../static/scripts/common/DBDefs';
-import {DONATE_URL} from '../../constants';
-import {bracketedText} from '../../static/scripts/common/utility/bracketed';
-import formatUserDate from '../../utility/formatUserDate';
+import {capitalize} from '../../static/scripts/common/utility/strings';
 import {returnToCurrentPage} from '../../utility/returnUri';
+import {CatalystContext} from '../../context';
+
+function languageName(language, selected) {
+  if (!language) {
+    return '';
+  }
+
+  const {
+    id,
+    native_language: nativeLanguage,
+    native_territory: nativeTerritory,
+  } = language;
+
+  let text = `[${id}]`;
+
+  if (nativeLanguage) {
+    text = capitalize(nativeLanguage);
+
+    if (nativeTerritory) {
+      text += ' (' + capitalize(nativeTerritory) + ')';
+    }
+  }
+
+  if (selected) {
+    text += ' \u25be';
+  }
+
+  return text;
+}
+
+const LanguageLink = ({$c, language}) => (
+  <a
+    href={
+      '/set-language/' + encodeURIComponent(language.name) +
+            '?' + returnToCurrentPage($c)
+    }
+  >
+    {languageName(language, false)}
+  </a>
+);
+
+type LanguageMenuProps = {
+  +$c: CatalystContextT,
+  +currentBCP47Language: string,
+  +serverLanguages: $ReadOnlyArray<ServerLanguageT>,
+};
 
-const Footer = (): React.Element<'div'> => {
+const LanguageMenu = ({
+  $c,
+  currentBCP47Language,
+  serverLanguages,
+}: LanguageMenuProps) => (
+  <div className="dropdown pb-1 mb-1 dropup">
+    <button
+      aria-expanded="false"
+      className="btn btn-outline-primary fs-4"
+      data-bs-toggle="dropdown"
+      id="languageDropdown"
+      type="button"
+    >
+      {languageName(
+        serverLanguages.find(x => x.name === currentBCP47Language),
+        true,
+      )}
+    </button>
+    <ul aria-labelledby="languageDropdown" className="dropdown-menu">
```

Comment:
We use kebab-case for class names and identifiers.

```suggestion
    <ul aria-labelledby="language-dropdown" className="dropdown-menu">
```

Reply:
RIght!